x-pack/plugins/spaces/server/spaces_client/spaces_client.test.ts

```diff
@@ -248,6 +254,79 @@ describe('#create', () => {
     });
     expect(mockCallWithRequestRepository.create).not.toHaveBeenCalled();
   });
+
+  describe('when config.allowFeatureVisibility is disabled', () => {
+    test(`creates space without disabledFeatures`, async () => {
+      const maxSpaces = 5;
+      const mockDebugLogger = createMockDebugLogger();
+      const mockCallWithRequestRepository = savedObjectsRepositoryMock.create();
+      mockCallWithRequestRepository.create.mockResolvedValue(savedObject);
+      mockCallWithRequestRepository.find.mockResolvedValue({
+        total: maxSpaces - 1,
+      } as any);
+
+      const mockConfig = createMockConfig({
+        enabled: true,
+        maxSpaces,
+        allowFeatureVisibility: false,
```

Comment:
Question: Why doesn't this fail the schema with 'Feature visibility can only be disabled on serverless'? Is it the context that is provided in the `createMockConfig`?

Reply:
Yeh, the `createMockConfig()` function automatically adds the correct serverless context depending on the value of `allowFeatureVisibility`